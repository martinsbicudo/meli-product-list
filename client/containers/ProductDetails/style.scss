.ProductDetails {
  padding: 2rem 0;
  display: flex;
  flex-direction: column;

  &,
  &CardBoard {
    flex: 1;
    width: 100%;
    box-sizing: border-box;
  }

  &CardBoard {
    display: flex;
    flex-wrap: wrap;

    > :last-child {
      width: 100%;
      margin-top: 5rem;
    }
  }

  &Figure {
    flex: 1;
    display: flex;
    align-items: flex-start;

    @include query-max("tablet") {
      min-width: 100%;
      justify-content: center;
      margin-bottom: 2rem;
    }

    @include query-min("tablet") {
      width: 60%;
      padding: 0 2rem 0 3rem;
    }

    img { max-width: 100%; }
  }

  &Title,
  &Price,
  &SubTitle,
  &Description {
    @include query-max("tablet") { text-align: center; }
  }

  &Infos {
    @include query-max("tablet") { width: 100%; }

    @include query-min("tablet") {
      width: 40%;
      padding: 5rem 3rem 0 0;
    }
  }

  &Title {
    font-size: 2.4rem;
    margin-bottom: 1rem;
  }

  &Price {
    @include query-max("tablet") { font-size: 2.4rem; }

    @include query-min("tablet") {
      font-size: 3rem;
      font-weight: bolder;
    }

    span {
      color: map-get($colors, "meli-dodger-blue");
      margin-right: .3rem;
    }
  }

  &Button {
    @include query-max("tablet") { margin: 3rem auto 0; }

    @include query-min("tablet") {
      width: 100%;
      margin-top: 3rem;
    }
  }

  &SubTitle {
    font-size: 2.4rem;
    margin-bottom: 2rem;
  }

  &Description { font-size: 1.2rem; }
}
