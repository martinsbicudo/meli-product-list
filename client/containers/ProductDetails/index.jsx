import React from 'react';
import { object } from 'prop-types';
import CardBoard from '@Components/CardBoard';
import Button from '@Components/Button';
import style from './style.scss';

const ProductDetails = ({
  product,
}) => (
  <div className={style.ProductDetails}>
    <CardBoard
      className={style.ProductDetailsCardBoard}
      footer={(
        <>
          <h2 className={style.ProductDetailsSubTitle}>Descrição do Produto</h2>
          <p className={style.ProductDetailsDescription}>{product.description}</p>
        </>
      )}
    >
      <figure className={style.ProductDetailsFigure}>
        <img
          src={product.pictures[0].url}
          alt={product.title}
          title={product.title}
        />
      </figure>
      <div className={style.ProductDetailsInfos}>
        <h1 className={style.ProductDetailsTitle}>
          {product.title}
        </h1>
        <p className={style.ProductDetailsPrice}>
          <span>$</span>
          {product.price}
          <Button
            className={style.ProductDetailsButton}
            modifiers={['dodger-blue', 'bigger']}
          >
            Comprar
          </Button>
        </p>
      </div>
    </CardBoard>
  </div>
);

ProductDetails.propTypes = {
  product: object.isRequired,
};

export default ProductDetails;
