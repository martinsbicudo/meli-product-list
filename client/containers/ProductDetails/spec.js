import ProductDetails from '.';

describe('Containers - <ProductDetails />', () => {
  const testMessageTitle = 'TEST_TITLE';
  const testMessagePrice = '300';
  const testMessageImage = 'TEST_IMAGE';
  const testMessageDescription = 'TEST_DESCRIPTION';
  const createWrapper = () => render(
    <ProductDetails
      product={{
        title: testMessageTitle,
        price: testMessagePrice,
        pictures: [{
          url: testMessageImage,
        }],
        description: testMessageDescription,
      }}
    />,
  );

  test('Should finds Title', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessageTitle)).toBeInTheDocument();
  });

  test('Should finds Price', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessagePrice)).toBeInTheDocument();
  });

  test('Should finds Image', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessagePrice)).toBeInTheDocument();
  });

  test('Should finds Description', () => {
    const { getByAltText } = createWrapper();

    expect(getByAltText(testMessageTitle).src).toBe(`http://localhost/${testMessageImage}`);
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
