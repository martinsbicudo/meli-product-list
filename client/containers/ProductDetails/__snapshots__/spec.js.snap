// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - <ProductDetails /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProductDetails"
  >
    <div
      class="CardBoard ProductDetailsCardBoard"
    >
      <figure
        class="ProductDetailsFigure"
      >
        <img
          alt="TEST_TITLE"
          src="TEST_IMAGE"
          title="TEST_TITLE"
        />
      </figure>
      <div
        class="ProductDetailsInfos"
      >
        <h1
          class="ProductDetailsTitle"
        >
          TEST_TITLE
        </h1>
        <p
          class="ProductDetailsPrice"
        >
          <span>
            $
          </span>
          300
          <button
            class="Button ProductDetailsButton ButtonDodgerBlue ButtonBigger"
            type="button"
          >
            Comprar
          </button>
        </p>
      </div>
      <div
        class="CardBoardFooter"
      >
        <h2
          class="ProductDetailsSubTitle"
        >
          Descrição do Produto
        </h2>
        <p
          class="ProductDetailsDescription"
        >
          TEST_DESCRIPTION
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
